services/system/ProxySys/src/ProxySys.cpp

```diff
@@ -79,10 +79,23 @@ namespace SystemService
       // TODO: avoid repacking (both directions)
       psibase::Actor<ServerInterface> iface(act.service, service);
 
+      auto reqTarget = req.target;
+
       auto result = iface.serveSys(std::move(req));
       if (result && !result->headers.empty() && serviceName != "common-sys")
          abortMessage("service " + service.str() + " attempted to set an http header");
 
+      if (reqTarget.size() == 0 || reqTarget.starts_with("/index.html"))
+      {
+         // Check if the target header is already in the vector; if the header was not found, add it
+         HttpHeader frameAncestorHeader = {"Content-Security-Policy", "frame-ancestors 'none';"};
+         auto pos = std::find(result->headers.begin(), result->headers.end(), frameAncestorHeader);
```

Comment:
This will miss many valid ways to spell the header.
Also, if the service has already provided a different value of `frame-ancestors`, do we still want to add `frame-ancestors 'none'`?